conda/common/configuration.py

```diff
@@ -1609,3 +1609,54 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def get_plugin_config_data(
+    data: dict[Path, dict[str, RawParameter]],
+) -> dict[Path, dict[str, RawParameter]]:
+    """
+    This is used to move everything under the key "plugins" from the provided dictionary
+    to the top level of the returned dictionary. The returned dictionary is then passed
+    to :class:`PluginConfig`.
+    """
+    new_data = {}
+
+    for source, config in data.items():
+        if plugin_data := config.get("plugins"):
+            for param_name, raw_param in plugin_data.value(None).items():
```

Comment:
What does `plugin_data.value(None)` do? Any way to have a nicer API that explains what that does?

Reply:
I looked at this closer. This assumes that we always get a dictionary back from the `.value()`, but that assumption is naive.

We need to add an extra check to make sure that the value we get back from that call some sort of mapping. If it isn't, we should skip it.